sdk_extensions/zpages/DESIGN.md

```diff
@@ -0,0 +1,175 @@
+# OpenTelemetry SDK Contrib - /tracez and /traceconfigz Design Doc
```

Comment:
There is a proposal for a docs folder here

https://github.com/open-telemetry/opentelemetry-java/issues/1397#issuecomment-655553297

I think for historical reference docs like design docs it could be nice to separate it away from the code. What do you think about moving it there?

Reply:
Sure, that sounds good to me. I'll rename the file to be more specific to /tracez.